exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/trace_exporter.rb

```diff
@@ -134,15 +134,17 @@ def send_bytes(bytes, timeout:) # rubocop:disable Metrics/AbcSize, Metrics/Cyclo
             retry_count = 0
             timeout ||= @timeout
             start_time = OpenTelemetry::Common::Utilities.timeout_timestamp
+            request_body = if @compression == 'gzip'
+                             Zlib.gzip(bytes)
+                           else
+                             bytes
+                           end
+
             around_request do # rubocop:disable Metrics/BlockLength
               request = Net::HTTP::Post.new(@path)
-              request.body = if @compression == 'gzip'
-                               request.add_field('Content-Encoding', 'gzip')
-                               Zlib.gzip(bytes)
-                             else
-                               bytes
-                             end
+              request.body = request_body
               request.add_field('Content-Type', 'application/x-protobuf')
+              request.add_field('Content-Encoding', 'gzip') if @compression == 'gzip'
               @headers.each { |key, value| request.add_field(key, value) }
```

Comment:
On retry, s it necessary to redo *any* of the work from the start of the block to here? Can we just hoist all of this out of the `around_request` block?

Reply:
I don't think so....I'd advocate moving it all out of the `around_request` block